samples/drivers/kscan/src/main.c

```diff
@@ -15,9 +15,13 @@
 
 LOG_MODULE_REGISTER(main);
 
+#ifdef CONFIG_KSCAN_MATRIX_DRIVER
```

Comment:
If you create dts/bindings/kscan/kscan-matrix.yaml for the KSCAN_MATRIX_DRIVER, then you won't need to change this sample.

Reply:
Considering that the kscan_matrix.c is a pseudo device, I am not sure where is a propriate DTS location to put this device node. That's the reason I created the device name in the Kconfig file.

Reply:
I see there is precedence for this approach from these devices:
`CONFIG_ETH_NATIVE_POSIX_DRV_NAME`
`CONFIG_CRYPTO_TINYCRYPT_SHIM_DRV_NAME`
`CRYPTO_MBEDTLS_SHIM_DRV_NAME`